src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -1105,10 +1107,12 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                         $message = ''
                                         $prefix = ''
 
-                                        # Don't show line information if script module
-                                        if (($myinv -and $myinv.ScriptName -or $myinv.ScriptLineNumber -gt 1 -or $err.CategoryInfo.Category -eq 'ParserError') -and !($myinv.ScriptName.EndsWith('.psm1', [System.StringComparison]::OrdinalIgnoreCase))) {
```

Comment:
I'm not sure I understand what the root cause of the referenced issue is. Can you please explain the bold part of the following statement from the PR description?

> but the cmdlet in the issue was returning an ErrorRecord with that category (which is allowed), but **the ConciseView script was only expecting an object that looked like what PowerShell would return**.

Essentially, you make the 'parser-error' check more restricted by adding the `$err.Exception` check in addition to the `Category` check (it's an `-and`). So, does that mean for the issue https://github.com/PowerShell/PowerShell/issues/19223, the remote error is rendered with line detailed information when it shouldn't?

Reply:
In the original issue, the cmdlet was setting the `Category` to `ParserError` (which is perfectly allowed), so then the formatting would just see this and expect it to be a PowerShell parser error and thus the known members to be available.  Since it wasn't a PowerShell parser error, it was hitting a null ref exception due to trying to use a member that doesn't exist.  Since within the time of execution of the script during formatting, we don't have a `ParseException` (see above), the proposed fix was to add an additional check of the type of exception.  The remoting aspect seems to due to how it re-wraps the error and then the expected members aren't available, but when explicitly setting `Category` to `ParserError` locally, it works fine.

Reply:
Thanks! This is much clearer!